mail_alias_creator/entry_processors/group.py

```diff
@@ -29,15 +29,14 @@ def process(self, alias_address_provider):
         """Process."""
         logger.debug("Processing group EP with {}".format(self.group))
         from ..main import LDAP
-        users = LDAP.get_users_in_group(self.group)
-        if users == []:
-            logger.warn("Group {} has no members.".format(self.group))
-        mails = LDAP.get_user_primary_mails(users)
-        for i, mail in enumerate(mails):
+
+        tuples = LDAP.get_uids_and_primary_mails_for_group(self.group)
+
+        for uid, mail in tuples:
             if mail is None:
-                logger.error("User {} does not exist or has no primary mail.".format(users[i]))
+                logger.error("User {} does not exist or has no primary mail.".format(uid))
                 if CONFIG["main"].getboolean("strict"):
                     exit(1)
             else:
```

Comment:
Yes in theory you are right.
But I moved the from `GroupEP` into the first lines of `get_uids_and_primary_mails_for_group`.
And I just noticed, that I used the old version there instead of the fixed one.
Will fix that.